### ISSUE TITLE
jScope won't display any data if there are NaN values

### ISSUE THREAD

**Issue by boxboy** [MDSplus/mdsplus#1796]
any version of mds+ on mac os after version 7-50-1 will not display any data if any of the datapoints are nan. It did work from that version going backwards

**Comment by zack-vii** [comment]
Can you provide a sample that shows the issue. Best an expression without tree dependency.

I tried for MdsDataProvider:
```
 Y: [0.0 : 0.4 : 0.1 , $ROPRAND , 0.6 : 1.0 : 0.1]
 X: [0.0 : 0.4 : 0.1 ,    0.5   , 0.6 : 1.0 : 0.1]
```
where $ROPRAND is the MDSplus representation of NaN.
It plots a line from (0., 0.) to (1., 1.) that is broken at (0.5,0.5).
This is what I would expect.

**Comment by boxboy** [comment]
So I looked a little closer and I am having this issue only with 2D data. So if the data is like:

  'node': '\\DTS03_TE',
  'data': array([[ nan,  nan,  nan,  nan,  nan],
         [ nan,  nan,  nan,  nan,  nan],
         [19.1, 21.6, 17.9, 26.4, 15.6],
         [ 7.2,  8.9, 11.5, 11.8,  7.3]]),
'dim0': array([0.001, 0.011, 0.021, 0.031]),
  'dim1': array([-0.10325, -0.05166, -0.00165,  0.04836,  0.10074]),

and the X,Y entries look like:

Y: data(\DTS03_TE)[1,*]
X: dim_of(\DTS03_TE,1)

And I have one of those for each of the array values, instead of getting 2 traces out of 4 (2 are all NaNs) none of them plot

**Comment by zack-vii** [comment]
So you expect it to plot
 Y: [[$ROPRAND], [$ROPRAND], [21.6], [8.9]]
 against
X: [-.10325,-.05166,-.00165,.04836,.10074]
which would look a bit like
```
 _________
|      \  |
|_._._._._|
```
Hm I got it to plot that but it required manual auto scaling and showed a buggy extra point that appears if you select a marker.

**Comment by boxboy** [comment]
Actually I was expecting it to plot:

Y: [19.1, 21.6, 17.9, 26.4, 15.6]
X: [-0.10325, -0.05166, -0.00165, 0.04836, 0.10074]

And then I have more entries for each row of the data array.

I also tried to do this in jScope directly:
Y:  [82.9,  88.6,  92.4,  84.6, 102. ,  96. ,  75.9,  83.5,  72.3,﻿71.9,  53.6,  44.5,  30.4,  23.,$ROPRAND,$ROPRAND ]
X: [-0.093, -0.053, -0.013,  0.026,  0.066,  0.106,  0.146,﻿0.186,0.22,  0.26,  0.30,  0.34,  0.39, 0.47,.5,.6]

That worked fine. But if the data are pulled from the mds+ sever via jscope they are not displayed.
I'll just note that I do not have this problem with older versions only anything beyond 7-50-1. We have also confirmed that the same issue exists in linux (as well as Mac)

**Comment by zack-vii** [comment]
could you instead try:
```
Y: data(\DTS03_TE)[,2]
X: dim_of(\DTS03_TE,1)
```
there might have been a change in row-maior vs column-maior. But even then, if you refer to the row index 1 you get the second row which is purely nan.

**Comment by boxboy** [comment]
no dice there.  I tried several iterations and didn't get anywhere

**Comment by GabrieleManduchi** [comment]
There is a bug that was recently introduced in 2D jScope visualization. 
In a couple of days I will issue a PR with the fix to that and others 
(missing error messages)


                             Gabriele

-- 
Gabriele Manduchi

Istituto Gas Ionizzati del CNR
Consorzio RFX - Associazione EURATOM/ENEA sulla Fusione
Corso Stati Uniti 4, 35127 Padova - Italy
ph +39-049-829-5039/-5000 fax +39-049-8700718
mailto:gabriele.manduchi@igi.cnr.it, http://www.igi.cnr.it

**Comment by zack-vii** [comment]
fixed by #1800
